lib/packwerk/configuration.rb

```diff
@@ -32,26 +36,51 @@ def from_packwerk_config(path)
     end
 
     DEFAULT_CONFIG_PATH = "packwerk.yml"
-    DEFAULT_INCLUDE_GLOBS = ["**/*.{rb,rake,erb}"]
-    DEFAULT_EXCLUDE_GLOBS = ["{bin,node_modules,script,tmp,vendor}/**/*"]
+    DEFAULT_INCLUDE_GLOBS = T.let(["**/*.{rb,rake,erb}"], T::Array[String])
+    DEFAULT_EXCLUDE_GLOBS = T.let(["{bin,node_modules,script,tmp,vendor}/**/*"], T::Array[String])
+
+    sig { returns(T::Array[String]) }
+    attr_reader(:include)
+
+    sig { returns(T::Array[String]) }
+    attr_reader(:exclude)
+
+    sig { returns(String) }
+    attr_reader(:root_path)
 
-    attr_reader(
-      :include, :exclude, :root_path, :package_paths, :custom_associations, :config_path, :cache_directory
-    )
+    sig { returns(T.any(String, T::Array[String])) }
+    attr_reader(:package_paths)
 
+    sig { returns(T::Array[Symbol]) }
+    attr_reader(:custom_associations)
+
+    sig { returns(T.nilable(String)) }
+    attr_reader(:config_path)
+
+    sig { returns(Pathname) }
+    attr_reader(:cache_directory)
+
+    sig do
+      params(
+        configs: T::Hash[T.untyped, T.untyped],
+        config_path: T.nilable(String),
+      ).void
+    end
     def initialize(configs = {}, config_path: nil)
-      @include = configs["include"] || DEFAULT_INCLUDE_GLOBS
-      @exclude = configs["exclude"] || DEFAULT_EXCLUDE_GLOBS
+      @include = T.let(configs["include"] || DEFAULT_INCLUDE_GLOBS, T.untyped)
+      @exclude = T.let(configs["exclude"] || DEFAULT_EXCLUDE_GLOBS, T.untyped)
       root = config_path ? File.dirname(config_path) : "."
-      @root_path = File.expand_path(root)
-      @package_paths = configs["package_paths"] || "**/"
-      @custom_associations = configs["custom_associations"] || []
-      @parallel = configs.key?("parallel") ? configs["parallel"] : true
-      @cache_enabled = configs.key?("cache") ? configs["cache"] : false
-      @cache_directory = Pathname.new(configs["cache_directory"] || "tmp/cache/packwerk")
+      @root_path = T.let(File.expand_path(root), String)
+      @package_paths = T.let(configs["package_paths"] || "**/", T.untyped)
+      @custom_associations = T.let(configs["custom_associations"] || [], T.untyped)
+      @parallel = T.let(configs.key?("parallel") ? configs["parallel"] : true, T.untyped)
```

Comment:
Could be `T::Boolean`

Reply:
I didn't want to strongly type some of the config values in the constructor because I think sorbet would complain before validation errors surface. I'll look into it before shipping because I don't remember after vacation.

Reply:
It just fails type checks somewhere else. I'll cast them to what they're supposed to be. If it isn't covered by our validation tests, we can fix it later.